src/Extension/KernelExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.DotNet.Interactive;
+using Microsoft.DotNet.Interactive.Commands;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Extension
+{
+    public static class KernelExtensions
+    {
+        public static T UseQuestionMagicCommand<T>(this T kernel, Evaluator evaluator) where T : Kernel
+        {
+            var cellIdArg = new Argument<int>("Question number");
+
+            var questionCommand = new Command("#!question", "This cell will be evaluated")
+            {
+                cellIdArg
+            };
+
+            questionCommand.Handler = CommandHandler.Create<int, KernelInvocationContext>((id, context) =>
+            {
+                context.OnComplete(async (invocationContext) =>
+                {
+                    if (invocationContext.Command is SubmitCode submitCode)
+                    {
+                        var evaluationCriteria = evaluator.GetInputExecutionCriterion(id);
+                        if (evaluationCriteria != "")
```

Comment:
It's preferable to use a custom type, or use a method like `bool TryFindEvaluationCriterion(out string code)`, rather than assigning semantic meaning to an empty string.

Reply:
Done